docs/agent-api.asciidoc

```diff
@@ -324,6 +324,14 @@ it's possible to use this option to supply an additional message string that wil
 
 ** `custom` - See <<metadata,metadata section>> for details about this option
 
+** `request` You can associate an error with an instance of `http.IncomingMessage` to gain additional context such as the request url, headers and cookies. However in most cases the agent will detect if an error was in response to a  http request and automatically add the request details for you. See <<http-requests, http requests section>> for more details.
+
+** `response` You can associate an error with an instance of `http.ServerResponse` to get additional details such as status code and headers. However in most cases the agent will detect if an error occured during a http request and automatically response details for you. See <<http-responses, http responses section>> for more details.
```

Comment:
Similar to above, this change suggestion requires that you add the following line to the `docs/index.asciidoc` file:

```asciidoc
:type-serverresponse: https://nodejs.org/api/http.html#http_class_http_serverresponse[<http.ServerResponse>]
```

Here's the complete change:

```suggestion
** `response` +{type-serverresponse}+ You can associate an error with information about the http response to get additional details such as status code and headers.
However, in most cases, the agent will detect if an error occurs during an http request and automatically response details for you.
See <<http-responses,http responses section>> for more details.
```